src/lib/io/Makefile

```diff
@@ -88,6 +88,10 @@ obj-io-storage-$(USE_EFIVARS) += \
     sol-efivarfs-storage.o
 obj-io-storage-$(USE_MEMMAP) += \
     sol-memmap-storage.o
+obj-io-storage-$(LINUX) += \
+    sol-memmap-storage-impl-linux.o
+obj-io-storage-$(ZEPHYR) += \
+    sol-memmap-storage-impl-zephyr.o
```

Comment:
should these be inside `if` blocks? This would include the zephyr or linux objects even if no `USE_STORAGE` is defined.